lib/spack/spack/test/build_environment.py

```diff
@@ -436,20 +436,13 @@ def _trust_me_its_a_dir(path):
     assert not (set(os.path.normpath(x) for x in link_dirs[:-2]) & external_lib_paths)
 
 
-def test_parallel_false_is_not_propagating(config, mock_packages):
-    class AttributeHolder(object):
-        pass
-
-    # Package A has parallel = False and depends on B which instead
-    # can be built in parallel
-    s = spack.spec.Spec("a foobar=bar")
-    s.concretize()
-
+def test_parallel_false_is_not_propagating(default_mock_concretization):
+    # Package A has parallel = False and depends on B which instead can be built in parallel
+    s = default_mock_concretization("a foobar=bar")
     for spec in s.traverse():
-        expected_jobs = spack.config.get("config:build_jobs") if s.package.parallel else 1
-        m = AttributeHolder()
-        spack.build_environment._set_variables_for_single_module(s.package, m)
-        assert m.make_jobs == expected_jobs
+        expected_jobs = spack.build_environment.determine_number_of_jobs(spec.package.parallel)
```

Comment:
This test should be written differently though, cause it's testing whether a property on the parent package is NOT set on the child (for whatever reason...), but now it queries the package properties doing that while verifying that

Reply:
I think the test was wrong before. If you look, we were iterating over nodes, but never using the nodes - we were always checking the result on the root.

Reply:
how about not iterating at all, it's really just two packages a and b